src/V3EmitCSyms.cpp

```diff
@@ -440,18 +444,27 @@ void EmitCSyms::emitSymHdr() {
     }
     puts("bool __Vm_didInit;\n");
 
-    puts("\n// SUBCELL STATE\n");
+    if (v3Global.opt.mtasks()) {
+        puts("VlThreadPool* const __Vm_threadPoolp;\n");
+        puts("bool __Vm_even_cycle;\n");
+
+        if (v3Global.opt.profThreads()) {
+            // rdtsc() at current cycle start
+            puts("vluint64_t __Vm_profile_cycle_start;\n");
+            // Time we finished analysis
+            puts("vluint64_t __Vm_profile_time_finished;\n");
+            // Track our position in the cache warmup and actual profile window
+            puts("vluint32_t __Vm_profile_window_ct;\n");
+        }
+    }
+
+    puts("\n// MODULE INSTANCE STATE\n");
     for (const auto& i : m_scopes) {
         AstScope* scopep = i.first;
         AstNodeModule* modp = i.second;
         if (VN_IS(modp, Class)) continue;
-        if (modp->isTop()) {
-            ofp()->printf("%-30s ", (prefixNameProtect(modp) + "*").c_str());
-            puts(protectIf(scopep->nameDotless() + "p", scopep->protect()) + ";\n");
-        } else {
-            ofp()->printf("%-30s ", (prefixNameProtect(modp) + "").c_str());
-            puts(protectIf(scopep->nameDotless(), scopep->protect()) + ";\n");
-        }
+        ofp()->printf("%-30s ", (prefixNameProtect(modp) + "").c_str());
```

Comment:
Shouldn't call c_str on temporary objects (possible bug with earlier code).  Add a string variable, then call on that.